solidity-v1/dashboard/src/pages/liquidity/LiquidityPage.jsx

```diff
@@ -181,36 +206,115 @@ const LiquidityPage = ({ headerTitle }) => {
           <Banner.CloseIcon onClick={hideBanner} />
         </Banner>
       )}
-
-      <KeepOnlyPool
-        {...KEEP_ONLY}
-        percentageOfTotalPool={KEEP_ONLY.shareOfPoolInPercent}
-        rewardBalance={KEEP_ONLY.reward}
-        addLpTokens={addLpTokens}
-        withdrawLiquidityRewards={withdrawLiquidityRewards}
-        liquidityContractName={LIQUIDITY_REWARD_PAIRS.KEEP_ONLY.contractName}
-        pool={LIQUIDITY_REWARD_PAIRS.KEEP_ONLY.pool}
-      />
-      <CardContainer>
-        {cards.map(({ id, ...data }) => (
-          <LiquidityRewardCard
-            key={id}
-            poolId={id}
-            {...data}
-            apy={liquidityPools[id].apy}
-            percentageOfTotalPool={liquidityPools[id].shareOfPoolInPercent}
-            rewardBalance={liquidityPools[id].reward}
-            wrappedTokenBalance={liquidityPools[id].wrappedTokenBalance}
-            lpBalance={liquidityPools[id].lpBalance}
-            lpTokenBalance={liquidityPools[id].lpTokenBalance}
-            rewardMultiplier={liquidityPools[id].rewardMultiplier}
-            isFetching={liquidityPools[id].isFetching}
-            addLpTokens={addLpTokens}
-            withdrawLiquidityRewards={withdrawLiquidityRewards}
-            isAPYFetching={liquidityPools[id].isAPYFetching}
-          />
-        ))}
-      </CardContainer>
+      <MasonryFlexContainer maxHeight={"2060px"}>
+        {cards.map(({ id, ...data }) => {
+          if (data.pool === POOL_TYPE.MSTABLE) {
```

Comment:
Let's add eg. `CardComponent` field to objects in `cards` array so we can get rid of `if` statements in map.
eg:
```
cards = [{ id: "KEEP_ETH", CardComponent: ActiveLiquidityRewardCard }]

cards.map({ id, CardComponent, ...data }) => <CardComponent key={id} poolId={id} {...data} />)
```